model/host/hostutil.go

```diff
@@ -175,3 +155,65 @@ func initSystemCommand() string {
 	fi
 	`
 }
+
+// FetchJasperCommands builds the command to download and extract the Jasper
+// binary into the directory dir.
+func (h *Host) FetchJasperCommand(config evergreen.JasperConfig, dir string) string {
+	return strings.Join(h.fetchJasperCommands(config, dir), " && ")
+}
+
+func (h *Host) fetchJasperCommands(config evergreen.JasperConfig, dir string) []string {
+	downloadedFile := h.jasperDownloadedFileName(config)
+	extractedFile := h.jasperExtractedFileName(config)
+	return []string{
+		fmt.Sprintf("cd \"%s\"", dir),
+		fmt.Sprintf("curl -LO '%s/%s'", config.URL, downloadedFile),
+		fmt.Sprintf("tar xzf '%s'", downloadedFile),
+		fmt.Sprintf("chmod +x '%s'", extractedFile),
+		fmt.Sprintf("rm -f '%s'", downloadedFile),
+	}
+}
+
+// FetchJasperCommandWithPath is the same as FetchJasperCommand but sets the
+// PATH variable to path for each command.
+func (h *Host) FetchJasperCommandWithPath(config evergreen.JasperConfig, dir, path string) string {
+	cmds := h.fetchJasperCommands(config, dir)
+	for i := range cmds {
+		cmds[i] = fmt.Sprintf("PATH=%s %s", path, cmds[i])
+	}
+	return strings.Join(cmds, " && ")
+}
+
+func (h *Host) jasperDownloadedFileName(config evergreen.JasperConfig) string {
+	os, arch := h.Distro.Platform()
+	return fmt.Sprintf("%s-%s-%s-%s.tar.gz", config.DownloadFileName, os, arch, config.Version)
+}
+
+func (h *Host) jasperExtractedFileName(config evergreen.JasperConfig) string {
+	if h.Distro.IsWindows() {
+		return config.BinaryName + ".exe"
+	}
+	return config.BinaryName
+}
+
+const (
+	windowsShell = "C:\\cygwin\\bin\\bash.exe"
```

Comment:
nit: you can use a raw string (backticks) to avoid the double slash situation. 

also it's not clear to me why some things are constants here and why some things are just string literals.

also I'd put the contants closer to the top of the file. 

also is there anyway we can avoid baking in or forcing the dependency between evergreen and cygwin?

Reply:
I added a distro setting to set the path to the shell so this doesn't depend on knowing cygwin exists.

The constants <powershell> and </powershell> are userdata-specific directives to tell cloud-init to run it as a powershell script.

/bin is necessary because Windows has its own path for PowerShell that needs to be overridden to be the cygwin /bin to prevent the user data script from using the wrong binary (e.g. Windows has a curl binary that behaves differently from the cygwin curl). I could also make the `PATH` variable a distro setting but at this point, there already seems to be a large proliferation of manual distro settings.